packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -122,6 +133,13 @@ export function PanelChrome({
           </TitleItem>
         </Tooltip>
       )}
+      {loadingState === LoadingState.Loading && onCancelQuery && (
+        <Tooltip content="Cancel query">
+          <TitleItem className={dragClassCancel} data-testid="panel-cancel-query" onClick={onCancelQuery}>
+            <Icon name="times-circle" size="md" />
+          </TitleItem>
+        </Tooltip>
+      )}
```

Comment:
I am torn on whether this should be here added as normal titleItem in PanelStateWrapper. 

:shrug: either should work, you decide :)

Reply:
I think maybe we should add the cancel button in a separate PR (it could need some tweaks to make it show only for slow queries). and if we add it here we should also remove it from the menu

Reply:
I will do that 😄

Reply:
[Done](https://github.com/grafana/grafana/pull/64796) ✔️